server/src/main/java/org/opensearch/index/translog/TranslogWriter.java

```diff
@@ -185,7 +185,7 @@ public static TranslogWriter create(
             if (Assertions.ENABLED) {
                 writerGlobalCheckpointSupplier = () -> {
                     long gcp = globalCheckpointSupplier.getAsLong();
-                    assert gcp >= initialGlobalCheckpoint : "global checkpoint ["
+                    assert gcp >= initialGlobalCheckpoint || gcp == -1 : "global checkpoint ["
```

Comment:
Just adding this condition does not seem right. In case remote translog is not enabled, then if later any bugs get introduced, they will be missed. We need to tie this with remote translog check.

Reply:
Sorry , I missed fixing this issue . Added `indexSettings` check now .